ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/MonadSTM/RAWLock.hs

```diff
@@ -144,6 +145,7 @@ import           Prelude hiding (read)
 -- * All public functions are exception-safe.
 --
 newtype RAWLock m st = RAWLock (StrictTVar m (RAWState st))
+  deriving NoThunks via OnlyCheckWhnfNamed "RAWLock" (RAWLock m st)
```

Comment:
Couldn't we newtype-derive `NoThunks` with an `IOLike m` constraint? (which implies `forall a. NoThunks a => StrictTVar m a`, and we have `NoThunks RAWState`)

Reply:
Hmmm, maybe?

Reply:
DOne